cmd/argocd/commands/plugin.go

```diff
@@ -0,0 +1,161 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type ArgoCDCLIOptions struct {
+	PluginHandler PluginHandler
+	Arguments     []string
+}
+
+// PluginHandler parses command line arguments
+// and performs executable filename lookups to search
+// for valid plugin files, and execute found plugins.
+type PluginHandler interface {
+	// LookForPlugin will iterate over a list of given prefixes
+	// in order to recognize valid plugin filenames.
+	// The first filepath to match a prefix is returned.
+	LookForPlugin(filename string) (string, bool)
```

Comment:
What you'd probably want is a way for `LookForPlugin` to be abstracted out so it can be more easily tested with `DefaultPluginHandler`. That could e.g be achieved by having `DefaultPluginHandler` have a field with `func(file string) (string, error)`, its default value being `exec.LookPath`.

`ExecutePlugin` can be abstracted in a similar way so that your tests don't need to invoke `cmd.Run`